src/api/routes/summaries.ts

```diff
@@ -136,49 +137,53 @@ export async function saveCrudeSummary(meta: CrudeSummaryDescriptor, summary: st
  */
 export async function findMissingCrudeSummaries(): Promise<CrudeSummaryDescriptor[]> {
   const ret: CrudeSummaryDescriptor[] = [];
-
-  const promises = (await listRecords())
-  // Only interested in meetings that are ready to download.
-  .filter(meeting => meeting.state === 3)
-  .map(async meeting => {
-    // Only interested in meetings that refers to valid groups.
-    const groupId = safeDecodeMeetingSubject(meeting.subject);
-    if (!groupId || !(await Group.count({ where: { id: groupId } }))) {
-      console.log(`Ignoring invalid meeting subject or non-existing group "${meeting.subject}"`)
-      return;
-    }
-
-    if (!meeting.record_files) return;
-    invariant(meeting.record_files.length == 1);
-    const startTime = meeting.record_files[0].record_start_time;
-    const endTime = meeting.record_files[0].record_end_time;
-
-    const record = await getRecordURLs(meeting.meeting_record_id);
-    const promises = record.record_files.map(async file => {
-      // Only interested in records that we don't already have.
-      const transcriptId = file.record_file_id;
-      if (await Summary.count({ where: {
-        transcriptId,
-        summaryKey: crudeSummaryKey,
-      } }) > 0) {
-        console.log(`Ignoring existing crude summaries for transcript "${transcriptId}"`)
-        return;
-      }
-
-      file.meeting_summary?.filter(summary => summary.file_type === 'txt')
-      .map(summary => {
-        ret.push({
-          groupId,
-          startedAt: startTime,
-          endedAt: endTime,
-          transcriptId,
-          url: summary.download_address,
+  const promises = [];
```

Comment:
don't change exiting logic or variable names. just wrap existing logic (including `await Promise.all(promises);`) in a for loop.